File: ProductofArrayExceptSelf.c
#include<stdio.h>
#include<stdlib.h>

int* productExceptSelf(int* nums, int numsSize, int* returnSize){
    int sum = 1;
    int sum2 = 1;
    int count = 0;
    int n = numsSize;
    for(int i = 0; i < n; ++i){
        sum *= nums[i];
        if(i != 0){
            sum2 *= nums[i];
        }else{
            count += 1;
        }
    }
    int *nums2 = malloc(n * sizeof(int));
    for(int i = 0; i < n; ++i){
        if(i == 0){
            if(count == 1){
                nums2[i] = sum2;
                continue;
            }else{
                nums2[i] = 0;
                continue;
            }
        }
        nums2[i] = sum/i;
    }
    return nums2;
}

File: ProductofArrayExceptSelf.py
def productExceptSelf(nums: list[int]) -> list[int]:
    product = 1
    count = 0
    for i in nums:
        if i == 0:
            count +=1
            if count > 1:
                return [0] * len(nums)
        else:
            product *= i
    lst = [0] * len(nums)
    for i in range(len(nums)):
        if nums[i] != 0 and count == 1:
            lst[i] == 0
        elif count == 0:
            lst[i] = int(product/i)
    return lst

print(productExceptSelf([-1,1,0,-3,3]))
